src/steamship/agents/context.py

```diff
@@ -0,0 +1,64 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Block
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class AgentContext(BaseModel, ABC):
```

Comment:
@dkolas where, in your estimation, do callbacks and monitoring hooks belong in this in this interface?

Reply:
It feels pretty comfortable for me to imagine them being in this object. In the good old days of wordpress, IIRC they just had a big flat namespace of hooks that you could even override even though some were "system" hooks.

We could have something similar -- a hook registry that the owning packageservice could even add hooks to -> or even hooks that called between agents of a team!